framework/wazuh/agent.py

```diff
@@ -2436,7 +2445,7 @@ def getconfig(self, component, configuration):
             return msg
         else:
             raise WazuhException(1101, rec_msg.replace("err ", ""))
```

Comment:
This is too generic.

If the error was "no such file or directory", you can return "Unable to get configuration due to the component is not running".
Otherwise, you can throw that exception.